### PR TITLE
Make some methods and fields public for inference

### PR DESCRIPTION
**Pull request by Ao-senXiong**
Related inference change: https://github.com/opprop/checker-framework-inference/pull/438

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ public static AnnotatedTypeMirror createType(
     protected final AnnotatedTypeFactory atypeFactory;
 
     /** Actual type wrapped with this AnnotatedTypeMirror. */
-    protected final TypeMirror underlyingType;
+    public final TypeMirror underlyingType;
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java:140]: There is a `getUnderlyingType()` getter method. Isn't it enough to access that?

File: framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java [hunk 2]
```diff
@@ -1243,8 +1243,7 @@ public void computeVarargType() {
          *
          * @param annotatedExecutableType an AnnotatedExecutableType
          */
-        /*package-private*/ void computeVarargType(
-                AnnotatedExecutableType annotatedExecutableType) {
+        public void computeVarargType(AnnotatedExecutableType annotatedExecutableType) {
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java:1246]: There is a public `computeVarargType` above.
Could that method be used instead? If not, can you add javadoc to make the relation clear?

**Review by wmdietl — COMMENTED** [review]
What is the relation between this PR and https://github.com/opprop/checker-framework-inference/pull/438 ?
I don't see a use of any of these changes, right? Is this PR obsolete?

**Comment by Ao-senXiong** [comment]
> What is the relation between this PR and [opprop/checker-framework-inference#438](https://github.com/opprop/checker-framework-inference/pull/438) ? I don't see a use of any of these changes, right? Is this PR obsolete?

Yes, sorry I should revert the change.

**Comment by Ao-senXiong** [comment]
Closed as the changes are not needed.

**Pull request closed without merging**